server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -299,6 +300,47 @@ public Metadata withIncrementedVersion() {
         );
     }
 
+    public Metadata withLifecycleState(final Index index, final LifecycleExecutionState lifecycleState) {
```

Comment:
Since this is a public method, can you add javadoc for it?

Reply:
How does ea9c861b31fbd129336acf1833b8455deed7336a look to you?